bpo-30125: disable faulthandler in ctypes test_SEH

vstinner opened the pull request:
Disable faulthandler to run test_SEH() of test_ctypes to prevent the
following log with a traceback:

    Windows fatal exception: access violation

vstinner committed:
bpo-30125: disable faulthandler in ctypes test_SEH

Disable faulthandler to run test_SEH() of test_ctypes to prevent the
following log with a traceback:

    Windows fatal exception: access violation

Add support.disable_faulthandler() context manager.

zware reviewed (approved):
LGTM, but when could faulthandler be missing?  I think you could use support.import_module to simplify things and still have protection from missing faulthandler if it's needed.

vstinner commented:
Wait... faulthandler.disable() is broken on Windows: I simply forgot to implement RemoveVectoredExceptionHandler()! I will write a second PR.

vstinner commented:
> LGTM, but when could faulthandler be missing?

Other Python implementations may implement ctypes but not faulthandler.

Note: PyPy implements faulthandler, but I don't know if they implemented the Windows exception handler.

vstinner commented:
See https://github.com/python/cpython/pull/1240 for the faulthandler.disable() fix.

vstinner commented:
Oh, the first run of test_ctypes using -W failed:

```
======================================================================
ERROR: test_SEH (ctypes.test.test_win32.FunctionCallTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\projects\cpython\lib\ctypes\test\test_win32.py", line 64, in test_SEH
    faulthandler.enable()
io.UnsupportedOperation: fileno
----------------------------------------------------------------------
Ran 465 tests in 0.922s
```

But the second run without -W succeeded.

I fixed my change to call enable() with file=sys.__stderr__.fileno() instead. I should now work in all cases. I also fixed enable() call: pass all_threads=True.

Since the code became more complex than expected, I moved it to a new helper in test.support.

I saw that test.support always import faulthandler, so I removed the try/except ImportError.
